test/GeometryInfoTestHelpers.h

```diff
@@ -259,6 +261,135 @@ inline auto geoPointOrWktMatcher = liftOptionalMatcher<GeoPointOrWkt>(
+  return AllOf(Property(&DPoint::getX,
+                        DoubleNear(expected.getX(), allowedCoordinateError)),
+               Property(&DPoint::getY,
+                        DoubleNear(expected.getY(), allowedCoordinateError)));
+};
+
+// ____________________________________________________________________________
+inline auto utilLineNear =
+    liftMatcherToElementsAreArray<DPoint, DLine>(utilPointNear);
+
+// ____________________________________________________________________________
+inline auto utilMultiPointNear =
+    liftMatcherToElementsAreArray<DPoint, DMultiPoint>(utilPointNear);
+
+// ____________________________________________________________________________
+inline auto utilMultiLineNear =
+    liftMatcherToElementsAreArray<DLine, DMultiLine>(utilLineNear);
+
+// ____________________________________________________________________________
+inline auto utilPolygonNear = [](DPolygon expected) -> Matcher<DPolygon> {
+  return AllOf(
+      Property(&DPolygon::getOuter, utilLineNear(expected.getOuter())),
+      Property(&DPolygon::getInners, utilMultiLineNear(expected.getInners())));
+};
+
+// ____________________________________________________________________________
+inline auto utilMultiPolygonNear =
+    liftMatcherToElementsAreArray<DPolygon, DMultiPolygon>(utilPolygonNear);
+
+// This helper simply calls `utilCollectionNear(expected)`. It is forward
+// declared here, s.t. `utilAnyGeometryNear` (on which the definition
+// `utilCollectionNear` depends) can use it recursively.
+Matcher<DCollection> utilCollectionNearForwardDecl(DCollection expected);
+
+// ____________________________________________________________________________
+inline auto utilAnyGeometryNear =
+    [](DAnyGeometry expected) -> Matcher<DAnyGeometry> {
+  using enum AnyGeometryMember;
+
+  auto geometryMatcher = [&]() -> ::testing::Matcher<DAnyGeometry> {
+    switch (AnyGeometryMember{expected.getType()}) {
+      case POINT:
+        return Property(&DAnyGeometry::getPoint,
+                        utilPointNear(expected.getPoint()));
+      case LINE:
+        return Property(&DAnyGeometry::getLine,
+                        utilLineNear(expected.getLine()));
+      case POLYGON:
+        return Property(&DAnyGeometry::getPolygon,
+                        utilPolygonNear(expected.getPolygon()));
+      case MULTILINE:
+        return Property(&DAnyGeometry::getMultiLine,
+                        utilMultiLineNear(expected.getMultiLine()));
+      case MULTIPOLYGON:
+        return Property(&DAnyGeometry::getMultiPolygon,
+                        utilMultiPolygonNear(expected.getMultiPolygon()));
+      case COLLECTION:
+        return Property(
+            &DAnyGeometry::getCollection,
+            utilCollectionNearForwardDecl(expected.getCollection()));
+      case MULTIPOINT:
+        return Property(&DAnyGeometry::getMultiPoint,
+                        utilMultiPointNear(expected.getMultiPoint()));
```

Comment:
If you make an overloaded struct `utilGeoNear`, then you can use the generic `visit` function here also.